task/npm/0.2/npm.yaml

```diff
@@ -0,0 +1,48 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: npm
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.17.0"
+    tekton.dev/categories: Build Tools
+    tekton.dev/tags: build-tool
+    tekton.dev/platforms: "linux/amd64,linux/s390x,linux/ppc64le"
+spec:
+  description: >-
+    This task can be used to run npm goals on a project.
+
+    This task can be used to run npm goals on a project
+    where package.json is present and has some pre-defined
+    npm scripts.
+  workspaces:
+    - name: source
+  params:
+    - name: PATH_CONTEXT
+      type: string
+      default: "."
+      description: The path where package.json of the project is defined.
+    - name: PACKAGE_MANAGER
+      type: string
+      default: "npm"
+      description: The package manager you want to run from the nodejs env.
+    - name: ARGS
+      type: array
+      default: ["version"]
+      description: The goals you want to run for the given command.
+    - name: IMAGE
+      type: string
+      default: "docker.io/library/node:12-alpine@sha256:dfbebf17bfb014e1e7068e76325a117bccf8679c68aec6a28514184a209c8bae"
+      description: The node image you want to use.
+  steps:
+    - name: npm-run
+      image: $(params.IMAGE)
+      command:
+        - $(params.PACKAGE_MANAGER)
```

Comment:
hey @waveywaves, I am bit confused over here :sweat_smile:. Instead of adding a new version of npm task should we add this as a new task in catalog since the current task name is `npm` which means that it will mostly be running `npm` commands and users might get confused. Any thoughts?

Reply:
I had the same thought initially, I am glad you are addressing it. 

I was initially going to create another task for yarn itself as I thought it might need another image which installs yarn. But, the nodejs image provides both yarn and npm as choices of package manager, hence it only made sense at the time to allow users to specify the package manager instead of creating an entirely different task for it. 

I guess it boils down to whether it makes sense to create another task for yarn or provide the flexibility for the user to not have to install an almost identical task and use the existing one by changing a param (which might be more usable).

Users might get confused only if the documentation around the param is not succinct enough or if they haven't read it. I can update the documentation accordingly if that is the change required also.

It's your call, I can create another PR for the yarn task if that is something that makes more sense to the catalog team.

Reply:
Hey @waveywaves 

Let me clear here, @vinamra28 is not asking for adding a new task for `yarn`. He is asking to add a new task maybe by the name of `nodejs` or something else which is flexible to use both `yarn` and `npm`. Adding the support for `yarn` in the task named `npm` does not look good. Also creating the two tasks doing the same thing is also not ideal.

I would say let's deprecate the `npm` task and add a new one with both `npm` and `yarn` support.

Reply:
Hi guys, can you also update the nodejs image in the existing npm task, it looks like node16 has better performance